test/controllers/UpfrontPaymentControllerSpec.scala

```diff
@@ -21,22 +21,105 @@ import play.api.mvc.Result
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import testsupport.ItSpec
+import testsupport.TdRequest.FakeRequestOps
+import testsupport.stubs.{AuthStub, EssttpBackend}
+import testsupport.testdata.TdAll
+import uk.gov.hmrc.http.SessionKeys
 
 import scala.concurrent.Future
 
 class UpfrontPaymentControllerSpec extends ItSpec {
 
   private val controller: UpfrontPaymentController = app.injector.instanceOf[UpfrontPaymentController]
 
-  "GET /upfrontpayment or whatever - Pawel, shall we add tests like these or just stick with int?" - {
-    "return 200 and the upfront payment page" ignore {
-      val fakeRequest = FakeRequest("GET", "/upfront-payment")
-      val result: Future[Result] = controller.upfrontPayment(fakeRequest)
+  private val serviceName: String = "Set up an Employers’ PAYE payment plan"
+  private val h1: String = "Can you make an upfront payment?"
+  private val pageContent: String =
+    "Your monthly payments will be lower if you can make an upfront payment. This payment will be taken from your bank account within 7 working days."
+  private val backLinkHtml: String = "<a href=\"/set-up-a-payment-plan/your-bill\" class=\"govuk-back-link\" id=\"back\">Back</a>"
+  private val signOutLinkHtml: String =
+    "<a class=\"govuk-link hmrc-sign-out-nav__link\" href=\"http://localhost:9949/auth-login-stub/session/logout\">\n              Sign out\n            </a>"
+
+
+  "GET /can-you-make-an-upfront-payment" - {
+    "return 200 and the can you make an upfront payment page" in {
+      AuthStub.authorise()
+      EssttpBackend.EligibilityCheck.findJourneyAfterEligibilityCheck
+
+      val fakeRequest = FakeRequest().withAuthToken().withSession(SessionKeys.sessionId -> "IamATestSessionId")
+
+      val result: Future[Result] = controller.canYouMakeAnUpfrontPayment(fakeRequest)
 
       status(result) shouldBe Status.OK
       contentType(result) shouldBe Some("text/html")
       charset(result) shouldBe Some("utf-8")
-      contentAsString(result) should include("Can you make an upfront payment?")
+
+      val pageContent: String = contentAsString(result)
+
+      pageContent should include(serviceName)
```

Comment:
we've got Jsoup already added as a dependency. We could have stronger tests here like:
```scala
val pageContent: String = contentAsString(result)
val doc = Jsoup.parse(pageContent)

doc.select("#back").attr("href") shouldBe routes.WhereEver.Itis.url
doc.select("form").attr("action") shouldBe routes.UpfrontPaymentControllerSpec.canPayUpfrontSubmit.url
doc.select(""selector-for-p1").text shouldBe "p1 text"
etc.
```

Reply:
Oh that's nicer, will implement